src/Orleans.Core/Runtime/AsynchAgent.cs

```diff
@@ -55,6 +55,7 @@ protected AsynchAgent(string nameSuffix, ExecutorService executorService, ILogge
 
             this.executorService = executorService;
             AppDomain.CurrentDomain.DomainUnload += CurrentDomain_DomainUnload;
+            EnsureExecutorInitialized();
```

Comment:
Is there a reason to initialize the executor before the call to Start?

Reply:
Partially constructed objects aren't the nicest entities, but for this case there's indeed no reason in such initialization. Removed.